sourcetracker/_sourcetracker.py

```diff
@@ -222,87 +222,64 @@ def subsample_dataframe(df, depth):
-        sink_data : np.array
-            Counts of each OTU in given sink.
-        num_sources : int
-            Number of sources, including unknown.
-
-        Attributes
-        ----------
-        sink_data : np.array
-            Sink data (see Parameters).
-        num_sources : int
-            Number of source environments.
-        num_features : int
-            Number of features.
-        sum : int
-            Total number of sequences in sink sample.
-        taxon_sequence : np.array
-            Bookkeeping array that contains an entry for each sequence so that
-            their source assignments can be manipulated. Not set until
-            `generate_taxon_sequence` is called.
-        seq_env_assignments : np.array
-            Vector of integers where the ith entry is the source environment
-            index of the ith sequence. Not set until
-            `generate_environment_assignments` is called.
-        envcounts : np.array
-            Total number of sequences assigned to each source environment. Not
-            set until `generate_environment_assignments` is called.
-        """
-        self.sink_data = sink_data.astype(np.int64)
-        self.sum = self.sink_data.sum()
-        self.num_sources = num_sources
-        self.num_features = sink_data.shape[0]
+    Parameters
+    ----------
+    sink_data : np.array
+        1D vector of ints. The ith column is the count of the ith feature.
 
-    def generate_taxon_sequence(self):
-        """Generate vector of taxa containing each sequence in self.sink_data.
+    Returns
+    -------
+    taxon_sequence : np.array
+        1D vector of ints.
 
-        Notes
-        -----
-        This function is used for book-keeping. The ith value of taxon_sequence
-        is the (integer) index of an OTU in self.sink_data. As an example, if
-        the sink data were [2, 0, 0, 4] (OTU_0 has an abundance of 2 and OTU_3
-        has an abundance of 4), the taxon sequence would be [0, 0, 3, 3, 3, 3].
-        """
-        self.taxon_sequence = np.zeros(self.sum, dtype=np.int32)
-        k = 0
-        for ind, num in enumerate(self.sink_data):
-            self.taxon_sequence[k: k + num] = ind
-            k += num
+    Notes
+    -----
+    This function creates a vector that allows the `gibbs_sampler` to
+    iteratively reassign sequences. For each feature in the `sink_data`, it
+    places k copies of feature i into the `taxon_sequence` vector. k is the
+    count of feature i in the sink data.
 
-    def generate_environment_assignments(self):
-        """Randomly assign each sequence in the sink to a source environment.
+    Examples
+    --------
+    >>> sink_data = np.array([4, 0, 0, 5, 2])
+    >>> ts = generate_taxon_sequence(sink_data)
+    >>> print(ts)
+    np.array([0, 0, 0, 0, 3, 3, 3, 3, 3, 4, 4])
+    '''
+    taxon_sequence = np.zeros(int(sink_data.sum()), dtype=np.int32)
+    k = 0
+    for ind, num in enumerate(sink_data.astype(np.int32)):
```

Comment:
I _think_ this can be replaced with `np.repeat`:

```python
21:00:07 (mcdonadt@Daniels-MacBook-Pro):~> a = np.arange(5)

21:00:10 (mcdonadt@Daniels-MacBook-Pro):~> np.repeat(a, [1, 0, 2, 1, 2])
Out[7]: array([0, 2, 2, 3, 4, 4])
```